databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/MeshOHazelcastPlugin.java

```diff
@@ -0,0 +1,129 @@
+package com.gentics.mesh.graphdb.cluster;
+
+import java.io.FileNotFoundException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.FileSystemXmlConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastInstanceNotActiveException;
+import com.orientechnologies.common.io.OFileUtils;
+import com.orientechnologies.common.log.OLogManager;
+import com.orientechnologies.common.parser.OSystemVariableResolver;
+import com.orientechnologies.common.util.OCallableNoParamNoReturn;
+import com.orientechnologies.common.util.OCallableUtils;
+import com.orientechnologies.orient.core.OSignalHandler;
+import com.orientechnologies.orient.core.Orient;
+import com.orientechnologies.orient.core.db.ODefaultEmbeddedDatabaseInstanceFactory;
+import com.orientechnologies.orient.server.OServer;
+import com.orientechnologies.orient.server.config.OServerParameterConfiguration;
+import com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedMap;
+import com.orientechnologies.orient.server.hazelcast.OHazelcastMergeStrategy;
+import com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin;
+import com.orientechnologies.orient.server.network.OServerNetworkListener;
+
+/**
+ * Custom Hazelcast plugin for OrientDB which supports injecting a previously instantiated hazelcast instance.
+ */
+public class MeshOHazelcastPlugin extends OHazelcastPlugin {
+
+	public static HazelcastInstance hazelcast;
+	public static Config hazelcastConfig;
+
+	@Override
+	public HazelcastInstance configureHazelcast() throws FileNotFoundException {
+		super.hazelcastInstance = hazelcast;
+		super.hazelcastConfig = hazelcastConfig;
+		return hazelcast;
+	}
+
+	public static HazelcastInstance createHazelcast(OServerParameterConfiguration[] iParams) throws FileNotFoundException {
+		if (hazelcast == null) {
+			String hazelcastConfigFile = null;
+			for (OServerParameterConfiguration param : iParams) {
+				if (param.name.equalsIgnoreCase("configuration.hazelcast")) {
+					hazelcastConfigFile = OSystemVariableResolver.resolveSystemVariables(param.value);
+					hazelcastConfigFile = OFileUtils.getPath(hazelcastConfigFile);
+				}
+			}
+
+			if (hazelcastConfigFile == null) {
+				throw new RuntimeException("Could not determine configuration setting.");
+			}
+
+			hazelcastConfig = new FileSystemXmlConfig(hazelcastConfigFile);
+			hazelcastConfig.setClassLoader(MeshOHazelcastPlugin.class.getClassLoader());
+			hazelcastConfig.getMapConfig(CONFIG_REGISTEREDNODES).setBackupCount(6);
+			hazelcastConfig.getMapConfig(OHazelcastDistributedMap.ORIENTDB_MAP).setMergePolicy(OHazelcastMergeStrategy.class.getName());
+			hazelcastConfig.setProperty("hazelcast.shutdownhook.enabled", "false");
```

Comment:
Why these settings? Remove them or explain them as a comment.

Reply:
Those settings have been copied from OHazelcast plugin. I'll add a comment on the shutdown hook. It is required to disable this since Mesh takes care of shutting down Hazelcast.